pkg/fanal/analyzer/language/nodejs/yarn/yarn.go

```diff
@@ -169,11 +169,54 @@ func (a yarnAnalyzer) parsePackageJsonDependencies(fsys fs.FS, path string) (map
 	}
 	defer func() { _ = f.Close() }()
 
-	pkg, err := a.packageJsonParser.Parse(f)
+	rootPkg, err := a.packageJsonParser.Parse(f)
 	if err != nil {
 		return nil, xerrors.Errorf("parse error: %w", err)
 	}
 
 	// Merge dependencies and optionalDependencies
-	return lo.Assign(pkg.Dependencies, pkg.OptionalDependencies), nil
+	dependencies := lo.Assign(rootPkg.Dependencies, rootPkg.OptionalDependencies)
+
+	if rootPkg.Workspaces != nil {
+		pkgs, err := a.traverseWorkspaces(fsys, rootPkg.Workspaces)
+		if err != nil {
+			return nil, xerrors.Errorf("traverse workspaces error: %w", err)
+		}
+		for _, pkg := range pkgs {
+			dependencies = lo.Assign(dependencies, pkg.Dependencies, pkg.OptionalDependencies)
+		}
+	}
+
+	return dependencies, nil
+}
+
+func (a yarnAnalyzer) traverseWorkspaces(fsys fs.FS, workspaces []string) ([]packagejson.Package, error) {
+	var pkgs []packagejson.Package
+
+	required := func(path string, _ fs.DirEntry) bool {
+		return filepath.Base(path) == types.NpmPkg
+	}
+
+	walkDirFunc := func(path string, d fs.DirEntry, r dio.ReadSeekerAt) error {
+		pkg, err := a.packageJsonParser.Parse(r)
```

Comment:
I think we need to add `name` and `version` of `package.json` file as dependency.
This is needed to build correct Dependency tree.

Reply:
From `package.json` we get only dependencies, in order to remove all indirect ones later. In order to build a correct tree, shouldn't we parse local dependencies, as is done [here](https://github.com/aquasecurity/go-dep-parser/pull/209/files#diff-74c6672ecb076a465ce5a8cb7cefa9fde706a90b13f616d88e3ff9978b65fe2f)?

Reply:
oh... right. It is yarn :smile:. 
We got dependencies from `yarn.lock` file.
Thank you for pointing that out!